README.md

```diff
@@ -1,93 +1,417 @@
+
+The latest release can always be found on the [releases page][github-releases].
+
+## Need help?
+
+If you run into problems using the SDK, you can
+
+* Ask questions on the [Okta Developer Forums][devforum]
+* Post [issues][github-issues] here on GitHub (for code errors)
+
+## Getting started
+
+Follow these steps to install the Okta Golang SDK in your project:
+- Run `go get github.com/okta/okta-sdk-golang`
+
+You'll also need
+
+* An Okta account, called an _organization_ (sign up for a free [developer organization](https://developer.okta.com/signup) if you need one)
+* An [API token](https://developer.okta.com/docs/api/getting_started/getting_a_token)
+
+Construct a client instance by passing it your Okta domain name and API token:
+
+```
+config := okta.NewConfig().WithOrgUrl("{yourOktaDomain}").WithToken("{apiToken}")
+client := okta.NewClient(config, nil, nil)
+```
+
+Hard-coding the Okta domain and API token works for quick tests, but for real projects you should use a more secure way of storing these values (such as environment variables). This library supports a few different configuration sources, covered in the [configuration reference](#configuration-reference) section.
+
+### Extending the Client
+When calling `okta.NewClient()` we allow for you to pass custom instances of `http.Client` and `cache.Cache`.
+
+```
+myClient := &http.Client{}
+
+myCache := NewCustomCacheDriver()
+
+config := okta.NewConfig().WithOrgUrl("{yourOktaDomain}").WithToken("{apiToken}")
+client := okta.NewClient(config, myClient, myCache)
+```
+
+
+### Extending or Creating New Cache Manager
+You can create a custom cache driver by implementing `cache.Cache`
+
+```
+type CustomCacheDriver struct {
+}
+
+func NewCustomCacheDriver() Cache {
+	return CustomCacheDriver{}
+}
+
+func (c CustomCacheDriver) Get(key string) *http.Response {
+	return nil
+}
+
+func (c CustomCacheDriver) Set(key string, value *http.Response) {}
+
+func (c CustomCacheDriver) Delete(key string) {}
+
+func (c CustomCacheDriver) Clear() {}
+
+func (c CustomCacheDriver) Has(key string) bool {
+	return false
+}
+```
+
+## Usage guide
+
+These examples will help you understand how to use this library. You can also browse the full [API reference documentation][sdkapiref].
+
+Once you initialize a `client`, you can call methods to make requests to the Okta API. Most methods are grouped by the API endpoint they belong to. For example, methods that call the [Users API](https://developer.okta.com/docs/api/resources/users) are organized under `client.User`.
+
+### Authenticate a User
+This library should be used with the Okta management API. To call the [Authentication API](https://developer.okta.com/docs/api/resources/authn), you should construct your own HTTP requests.
+
+### Get a User
+```
+config := okta.NewConfig()
+client := okta.NewClient(config, nil, nil)
+user, resp, err := client.User.GetUser(user.Id, nil)
+```
+
+### List all Users
+```
+config := okta.NewConfig()
```

Comment:
No need to repeat client instantiation in each example - want to move it above `### Authenticate a user`? I should update my template to reflect this.

Reply:
Personally, I like that it has this in every example... makes it more copy/pasteable?